timings/timings.go

```diff
@@ -80,6 +80,14 @@ func New(tags map[string]string) *Timings {
 	}
 }
 
+// SetTag sets a tag on the Timings object.
+func (t *Timings) SetTag(tag, value string) {
```

Comment:
I would call this AddTag even  if we possibly don't check for clashes,   the typical use case is to add tags not set existing ones.